drivers/can/stm32_can.c

```diff
@@ -888,4 +890,71 @@ static void config_can_1_irq(CAN_TypeDef *can)
 	can->IER |= CAN_IT_TME | CAN_IT_ERR | CAN_IT_FMP0 | CAN_IT_FMP1;
 }
 
+#if defined(CONFIG_NET_SOCKETS_CAN)
+
+#define SEND_TIMEOUT K_MSEC(100)
+#define SOCKET_CAN_NAME_1 "SOCKET_CAN_1"
+
+struct socket_can_context {
+	struct device *can_dev;
+
+	/* Transport protocol class address */
+	struct can_tp_addr can_addr;
+};
+
+static void socket_can_iface_init(struct net_if *iface)
+{
+	struct device *dev = net_if_get_device(iface);
+	struct socket_can_context *context = dev->driver_data;
+
+	LOG_DBG("Init CAN interface %p", iface);
+
+	net_if_set_link_addr(iface, (u8_t *)&context->can_addr,
+			     sizeof(context->can_addr),
+			     NET_LINK_CANBUS);
+}
+
+static int socket_can_send(struct device *dev, struct net_pkt *pkt)
+{
+	struct socket_can_context *socket_context = dev->driver_data;
+	int ret;
+
+	ret = can_send(socket_context->can_dev,
+		       (struct can_msg *)pkt->frags->data,
+		       SEND_TIMEOUT, NULL);
+	if (!ret) {
+		LOG_DBG("Cannot send socket CAN msg (%d)", ret);
+	}
+
+	return ret;
+}
+
```

Comment:
socket_can_send and socket_can_iface_init looks very generic. Could this be offloaded to socketcan.c ?
Maybe we get more than one driver implementation in future :-)

Reply:
Yes, that can be done.